packages/react/src/components/CodeSnippet/CodeSnippet.js

```diff
@@ -114,6 +115,15 @@ function CodeSnippet({
     handleScroll();
   }, [handleScroll]);
 
+  const handleCopyClick = (evt) => {
```

Comment:
@tw15egan
@jnm2377

If I read handleCopyClick  correctly it will either...
- call onClick
- or perform a copy-to-clipboard

Please consider our requirement that the copy-to-clipboard is done **`AND`** the onClick is called for additional processing.

My comment in #4448

I thank you two for continuing to address this.
We - IBM Cognos - have also ran into this issue and will benefit from the fix.

I would like to make a request.
In @repjarms's comment on Jan 30 the it was mentioned to override the onClick - and I agree this is needed.
But I think there's also a case for not overriding but to propagate the onClick.

Usecase (we have this situation):

client calls CodeSnippet with an onClick
client wants the copy-to-clipboard to happen and also receive an onClick event to do some other task
I'm not sure the best way to accomplish this.

Maybe an additional prop(copyToClipBoard) to indicate if the copy-to-clipboard should happen (or not happen).

 ```
const handleCopyClick = (evt) => {
    if( !onClick || this.props.copyToClipBoard) {
        copy(children);
    }
    onClick?.(evt);
};
```

Reply:
`onClick` will now be propagated after the copy function is executed

Reply:
I hate to bring it up but....
What if we don't want it to copy when supplying an onClick?
The client wants to handle it themself and/or does not want to pollute/overwrite what is in the clipboard.

Reply:
Discussed with TJ.
There may be a need to deal with actions besides copy.
This is ok for now.